### PR TITLE
Minor orderbook corrections to ensure empty orders are not handed to the clearing algorithm

### PR DESCRIPTION
**Pull request by adamsjohanna**
The complex clearing does not work, if orders have a volume of 0. Since those orders are not necessary in any case (that I'm aware of), those empty orders are deleted before handing over the orderbook to the clearing mechanism.

### CODE REVIEW

File: assume/markets/clearing_algorithms/complex_clearing.py [hunk 1]
```diff
@@ -217,11 +217,16 @@ def primal_eql_dual(m):
     eps = 1e-4
     solvers = check_available_solvers(*SOLVERS)
 
-    solver = SolverFactory(solvers[0])
+    solver = (
+        SolverFactory("gurobi") if "gurobi" in solvers else SolverFactory(solvers[0])
```
Comment (maurerle) [inline assume/markets/clearing_algorithms/complex_clearing.py:221]: If you want to prefer gurobi, you can reorder the list of Solvers at the top of the file ;)
I think it makes sense to prioritize commercial solvers if you have them available anyway.

File: assume/markets/clearing_algorithms/complex_clearing.py [hunk 2]
```diff
@@ -217,11 +217,16 @@ def primal_eql_dual(m):
...
+    )
     if solver.name == "gurobi":
         options = {"cutoff": -1.0, "eps": eps}
     else:
         options = {}
+        raise ValueError(
+            f"Solver {solver} does not support cutoff, might not find solution."
```
Comment (maurerle) [inline assume/markets/clearing_algorithms/complex_clearing.py:228]: in other cases, setting a timelimit might be good, which returns the best solution found

**Review by maurerle — COMMENTED** [review]
Maybe you can fix the SOLVERS list, other than that, it is fine

File: assume/markets/clearing_algorithms/complex_clearing.py (hunk 2 above)
Comment (adamsjohanna) [reply assume/markets/clearing_algorithms/complex_clearing.py:228]: I'm not that familiar with all the solver options, but yea, it should be possible to run it with different solvers. Maybe we could also implement the option to choose solver settings in the configs? At least the time limit could be a nice parameter to enter manually.

**Review by maurerle — APPROVED** [review]

**Pull request merged**